### PR TITLE
fix python bug and add java example

### PR DESCRIPTION
**Pull request by HexToString**
1.with open is python3 function, not fit for python2.x, change it with io.open, which is fit both for python2.x and python3.
2.add java pipeline client example.（which is also fit for mutil-thread）
3.in order to keep the original python pipeline example complete, add python pipeline web_service_java.py as a pipeline server for java pipeline client.
4.add some information in java README.
5.add some annotation.

### CODE REVIEW

File: python/examples/pipeline/simple_web_service/web_service_java.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import logging
+import numpy as np
+from numpy import array
+import sys
+import base64
+
+_LOGGER = logging.getLogger()
+np.set_printoptions(threshold=sys.maxsize)
+class UciOp(Op):
+    def init_op(self):
```
Comment (TeslaZhao) [inline python/examples/pipeline/simple_web_service/web_service_java.py:27]: 为什么要新增一个文件web_service_java.py，用web_service.py不行吗？

Comment (HexToString) [reply python/examples/pipeline/simple_web_service/web_service_java.py:27]: 不可以，由于restful 请求的格式，input 类型是list，必须在web_service.py中转换为np.
web_server_java.py，java client input 类型直接是INDArray，不需要进行转换。

**Comment by HexToString** [comment]
reopen it.

**Comment by HexToString** [comment]
reopen it and add some annotation

**Review by TeslaZhao — APPROVED** [review]
LGTM

**Pull request merged**